pilot/pkg/xds/gen2.go

```diff
@@ -15,16 +15,24 @@
 package xds
 
 import (
+	"fmt"
 	"time"
 
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 	"google.golang.org/grpc/codes"
 
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/pkg/env"
+	istioversion "istio.io/pkg/version"
 )
 
 // gen2 provides experimental support for extended generation mechanism.
 
+var (
+	podName = env.RegisterStringVar("POD_NAME", "", "").Get()
```

Comment:
Can we pass this in from pilotargs maybe?

Reply:
@howardjohn Any clue how to get it?  I could easily add a new field in Env or Server, but I couldn't find a reference from DiscoveryServer back to PilotArgs?